woocommerce/woocommerce-android
# Composite products hierarchy in order details

Closes: #8406

### Why 
On the Eagle team, we are working on improving the experience for our users across our Canonical Extensions. One of these extensions is Composite Products, and one of our tasks and ideas to improve the experience was to display the relationship between a composite product and its components.

### Description
This PR uses the new parent field on the Order class to group all components inside its composite product parent.

### Testing instructions
#### Prerequisites
1. Create a composite product containing products on wp-admin
2. Place an order containing at least one composite product

#### Test case
1. Open the Orders list
2. Open an order containing a composite product
3. Check that the composite product's components appear with more margin than the parent (composite product)

### Images/gif
https://github.com/woocommerce/woocommerce-android/assets/18119390/838c65e9-0337-4aa6-80aa-bce7c615c16f
